File: solutions/r/raindrops/1/raindrops.R
raindrops <- function(n) {
  result <- c(
    if (n %% 3 == 0) "Pling" else NULL,
    if (n %% 5 == 0) "Plang" else NULL,
    if (n %% 7 == 0) "Plong" else NULL
  )

  if (length(result) == 0) {
    return(as.character(n))
  } else {
    return(paste(result, collapse = ""))
  }
}

File: solutions/clojure/raindrops/1/src/raindrops.clj
(ns raindrops
  (:require [clojure.string :as str]))

(defn- div-by? [n d] (zero? (rem n d)))
(defn convert
  [num]
  (if-let [result (not-empty (str/join [(when (div-by? num 3) "Pling")
                                        (when (div-by? num 5) "Plang")
                                        (when (div-by? num 7) "Plong")]))]
    result
    (str num)))
